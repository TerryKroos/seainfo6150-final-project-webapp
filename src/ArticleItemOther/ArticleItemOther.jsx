import React from 'react';
import styles from "./ArticleItemOther.module.css";
import {BrowserRouter as Router, Link} from 'react-router-dom';
import { useHistory } from "react-router-dom";


const ArticleItemOther = (props)=>{
    
    const history = useHistory();
    const handleSubmit = () => {
        history.push('/article', {params: props.article})
    }
    return(
        <div className={styles.container} onClick={handleSubmit}>
           <div className={styles.imagediv}>
              <img className ={styles.image}src={props.article.image} alt="article img"/>
           </div>
           <div className={styles.articletext}>
               <Router>
              <Link className={styles.link}>
              {props.article.heading}</Link>
              </Router>
              <time datetime="2018-11-22">{props.article.date}</time>
           </div>
        </div>
        );
}

export default ArticleItemOther;