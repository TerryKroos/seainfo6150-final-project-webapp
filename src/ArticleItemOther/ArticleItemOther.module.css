.container {
    display: flex;
    flex-direction: row;
    height: 7em;
    width: 60em;
    background-color: white;
    color:#556575;
    cursor: pointer;
    margin-bottom: 20px;
    border: 2px solid #F0F0F0;
    border-radius: 10px 10px 10px 10px; 
}
.imagediv {
    width: 20em;

}
.image {
    border-radius: 10px 0px 0px 10px; 
 width: 100%;   
 height: 100%;

}
.articletext {
    display: flex;
    flex-direction: column;
    justify-content: center;
    margin-left: 10px;
    height: 100%;
    width:100%;
    background-color: white;
}
.link{
    text-decoration: none;
    color: #556575;
    cursor: pointer;
  }
  .link:hover{
    color: #282828; 
  }

@media (min-width: 300px){  
    .container {
        display: flex;
        flex-direction: row;
        height: 7em;
        width: 30em;
        background-color: white;
        cursor: pointer;
        margin-bottom: 20px;
        border: 2px solid #F0F0F0;
        border-radius: 10px 10px 10px 10px; 
        font-size: 13px;
    }
    time {
        font-size: 11px;
    }
    
    }


@media (min-width: 600px){  
    .container {
       
        height: 7em;
        width: 40em;

        
    }
    
    }

    @media (min-width: 900px){  
        .container {
            height: 7em;
            width: 60em;
            font-size: 15px;
        }
        time {
            font-size: 13px;
        }
        
        }