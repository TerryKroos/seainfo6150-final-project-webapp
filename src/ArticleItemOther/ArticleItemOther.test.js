
import React from "react";
import ArticleItemOther from "./ArticleItemOther";
import { render } from "@testing-library/react";

describe("ArticleItemOther tests", () => {
    it("renders correctly", () => {
      const { container } = render(<ArticleItemOther article={{}}/>);
      expect(container).toMatchSnapshot();
    });
  
  });