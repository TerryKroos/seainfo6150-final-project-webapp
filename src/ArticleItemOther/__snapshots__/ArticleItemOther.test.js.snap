// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleItemOther tests renders correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="imagediv"
    >
      <img
        alt="article img"
        class="image"
      />
    </div>
    <div
      class="articletext"
    >
      <a
        class="link"
        href=""
      />
      <time
        datetime="2018-11-22"
      />
    </div>
  </div>
</div>
`;
